pkg/infra/tracing/tracing_test.go

```diff
@@ -5,137 +5,149 @@ import (
-	defer func() {
-		err := os.Unsetenv("JAEGER_DISABLED")
-		require.NoError(t, err)
-	}()
-
-	ts := &Opentracing{enabled: true}
-	cfg, err := ts.initJaegerCfg()
-	require.NoError(t, err)
-
-	assert.True(t, cfg.Disabled)
-}
-
-func TestInitJaegerCfg_EnabledViaEnv(t *testing.T) {
-	err := os.Setenv("JAEGER_DISABLED", "false")
-	require.NoError(t, err)
-	defer func() {
-		err := os.Unsetenv("JAEGER_DISABLED")
-		require.NoError(t, err)
-	}()
-
-	ts := &Opentracing{enabled: false}
-	cfg, err := ts.initJaegerCfg()
-	require.NoError(t, err)
-
-	assert.False(t, cfg.Disabled)
-}
-
-func TestInitJaegerCfg_InvalidEnvVar(t *testing.T) {
-	err := os.Setenv("JAEGER_DISABLED", "totallybogus")
-	require.NoError(t, err)
-	defer func() {
-		err := os.Unsetenv("JAEGER_DISABLED")
-		require.NoError(t, err)
-	}()
-
-	ts := &Opentracing{}
-	_, err = ts.initJaegerCfg()
-	require.EqualError(t, err, "cannot parse env var JAEGER_DISABLED=totallybogus: strconv.ParseBool: parsing \"totallybogus\": invalid syntax")
-}
-
-func TestInitJaegerCfg_EnabledViaHost(t *testing.T) {
-	require.NoError(t, os.Setenv("JAEGER_AGENT_HOST", "example.com"))
-	defer func() {
-		require.NoError(t, os.Unsetenv("JAEGER_AGENT_HOST"))
-	}()
-
-	cfg := setting.NewCfg()
-	ts := &Opentracing{Cfg: cfg}
-	_, err := ts.Cfg.Raw.NewSection("tracing.jaeger")
-	require.NoError(t, err)
-	require.NoError(t, ts.parseSettings())
-	jaegerCfg, err := ts.initJaegerCfg()
-	require.NoError(t, err)
+func TestSplitCustomAttribs_Malformed(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected []attribute.KeyValue
+	}{
+		{input: "key1=value1"},
+		{input: "key1"},
+	}
 
-	assert.False(t, jaegerCfg.Disabled)
-	assert.Equal(t, "example.com:6831", jaegerCfg.Reporter.LocalAgentHostPort)
+	for _, test := range tests {
+		_, err := splitCustomAttribs(test.input)
+		assert.Error(t, err)
+	}
 }
 
-func TestInitJaegerCfg_EnabledViaHostPort(t *testing.T) {
-	require.NoError(t, os.Setenv("JAEGER_AGENT_HOST", "example.com"))
-	require.NoError(t, os.Setenv("JAEGER_AGENT_PORT", "12345"))
-	defer func() {
-		require.NoError(t, os.Unsetenv("JAEGER_AGENT_HOST"))
-		require.NoError(t, os.Unsetenv("JAEGER_AGENT_PORT"))
-	}()
-
-	cfg := setting.NewCfg()
-	ts := &Opentracing{Cfg: cfg}
-	_, err := ts.Cfg.Raw.NewSection("tracing.jaeger")
-	require.NoError(t, err)
-	require.NoError(t, ts.parseSettings())
-	jaegerCfg, err := ts.initJaegerCfg()
-	require.NoError(t, err)
-
-	assert.False(t, jaegerCfg.Disabled)
-	assert.Equal(t, "example.com:12345", jaegerCfg.Reporter.LocalAgentHostPort)
+func TestTracingConfig(t *testing.T) {
+	for _, test := range []struct {
+		Name               string
+		Cfg                string
+		Env                map[string]string
+		ExpectedExporter   string
+		ExpectedAddress    string
+		ExpectedPropagator string
+		ExpectedAttrs      []attribute.KeyValue
+	}{
+		{
+			Name:             "default config uses noop exporter",
+			Cfg:              "",
+			ExpectedExporter: "noop",
```

Comment:
nit, check for the [constant](https://github.com/grafana/grafana/blob/a68c9d0616a5b7b90fbf4274a20297ab7664659b/pkg/infra/tracing/tracing.go#L42) instead